### PR TITLE
Docs website change: Pydata docs theme

### PR DESCRIPTION
**Pull request by zain-sohail**
This PR changes the current readthedocs theme we are using for docs to the pydata docs theme, which supports a lot more features.
Here, we also change the documentation github actions workflow as after building, it will push that versions (like 0.2 etc.) to the docs repo in the sed folder. 

The latest main version can be found here: https://opencompes.github.io/docs/sed/latest/

There are some changes to the structure of the documentation (such as seperating user guide, advanced workflows and api) that I saw made more sense but these can be discussed here.

Current main branch is supporting python 3.8 but the pydata theme only supports >3.9 so the workaround is to pip install on the CI for now but in v1, we can change this.

### CODE REVIEW

File: .cspell/custom-dictionary.txt [hunk 1]
```diff
@@ -403,3 +403,7 @@ pypi
 pypa
 cobertura
 coverallsapp
+viewcode
+pydata
+ALLUSERPROFILE
```
Comment (rettigl) [inline .cspell/custom-dictionary.txt:408]: This is a typo, must be ALLUSERSPROFILE

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -84,25 +66,63 @@ jobs:
...
+          path: sphinx-docs
+
+      - name: Determine version folder
+        id: version-folder
+        run: |
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            VERSION=${GITHUB_REF#refs/tags/}
+            echo "folder=sed/$VERSION" >> $GITHUB_OUTPUT
+          else
+            echo "folder=sed/latest" >> $GITHUB_OUTPUT
```
Comment (rettigl) [inline .github/workflows/documentation.yml:113]: This will alyways set any manual run of this workflow as the latest version. I suggest to add a case for generating docs for specific branches for testing.

File: .github/workflows/documentation.yml [hunk 3]
```diff
@@ -84,25 +66,63 @@ jobs:
...
+          fi
+
+      - name: Copy documentation to the right version folder
+        run: |
+          mkdir -p docs-repo/${{ steps.version-folder.outputs.folder }}
+          cp -r sphinx-docs/* docs-repo/${{ steps.version-folder.outputs.folder }}
```
Comment (rettigl) [inline .github/workflows/documentation.yml:119]: This contains a substantial amount of unnecceary data (>50 mbyte per generated docs version). One could e.g. omitt the .doctrees folder, and the ipynb files from the tutorials folder.

File: docs/conf.py [hunk 4]
```diff
@@ -98,7 +101,28 @@ def _get_project_meta():
...
+    "navbar_center": ["version-switcher", "navbar-nav"],
+    "show_nav_level": 2,
+    "show_version_warning_banner": True,
+    # maybe better to use _static/switcher.json on github pages link instead of the following
+    "switcher": {
+        "json_url": "https://raw.githubusercontent.com/OpenCOMPES/docs/main/sed/switcher.json",
+        "version_match": version,
+    },
+    "content_footer_items": ["last-updated"],
+}
```
Comment (rettigl) [inline docs/conf.py:118]: I don't like the curent layout so much, which does not show the whole docs tree in the left sidebar (e.g., if you are in tutorials, you don't see the main menu entries). can this be changed to look similar to the old theme?

File: docs/user_guide/config.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+It will load an (optional) provided config file, or alternatively use a passed python dictionary as initial config dictionary, and subsequently look for the following additional config files to load:
+
+* ``folder_config``: A config file of name :file:`sed_config.yaml` in the current working directory. This is mostly intended to pass calibration parameters of the workflow between different notebook instances.
+* ``user_config``: A config file provided by the user, stored as :file:`.sed/config.yaml` in the current user's home directly. This is intended to give a user the option for individual configuration modifications of system settings.
+* ``system_config``: A config file provided by the system administrator, stored as :file:`/etc/sed/config.yaml` on Linux-based systems, and :file:`%ALLUSERPROFILE%/sed/config.yaml` on Windows. This should provide all necessary default parameters for using the sed processor with a given setup. For an example for an mpes setup, see :ref:`example_config`
```
Comment (rettigl) [inline docs/user_guide/config.md:8]: ALLUSERSPROFILE
Why did you introduce this typo here again?

File: docs/user_guide/index.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+:caption: Installing SED
+:maxdepth: 1
+installation
+```
+
+```{toctree}
+:caption: Basic concepts
+../tutorial/1_binning_fake_data
+../tutorial/8_jittering_tutorial
+../tutorial/3_metadata_collection_and_export_to_NeXus
```
Comment (rettigl) [inline docs/user_guide/index.md:19]: I don't find these helpful here in this order. They hardly can be understood here without the other tutorials.

File: docs/user_guide/index.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+```
+
+```{toctree}
+:caption: Advance
+config
```
Comment (rettigl) [inline docs/user_guide/index.md:24]: Why is config advanced? This should be in the basic concepts

Comment (rettigl) [reply docs/user_guide/index.md:24]: Explanation for Jittering is maybe more something advanced

File: docs/user_guide/installation.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+.sed-dev\Scripts\activate
+```
+
+3. Install the repository in editable mode with all dependencies:
+
+```bash
+pip install -e .[all]
+```
+
+Now you have the development version of `sed` installed in your local environment. Feel free to make changes and submit pull requests.
```
Comment (rettigl) [inline docs/user_guide/installation.md:74]: Why did you remove poetry docs?

File: docs/workflows/index.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+      Workflows showcasing different SED loaders and methods
+---
+# Workflows
+
+```{toctree}
+:caption: Basic concepts
+../tutorial/2_conversion_pipeline_for_example_time-resolved_ARPES_data
+../tutorial/4_hextof_workflow
+../tutorial/6_binning_with_time-stamped_data
+../tutorial/7_correcting_orthorhombic_symmetry
```
Comment (rettigl) [inline docs/workflows/index.md:14]: This separation does not really make sense to me.

**Review by rettigl — CHANGES_REQUESTED** [review]
I generally like the idea behind the new workflow, but suggest some improvements to the structure and layout. See detailed comments.

File: docs/user_guide/config.md (hunk 5 above)
Comment (zain-sohail) [reply docs/user_guide/config.md:8]: I think I included the typo in cspell dict and it propogated it here

File: docs/conf.py (hunk 4 above)
Comment (zain-sohail) [reply docs/conf.py:118]: It should be possible to change that, I believe. However, those entries are showing on the right side, which I find quite a cleaner layout. And I can directly navigate to other tutorials without issue.

File: docs/user_guide/index.md (hunk 7 above)
Comment (zain-sohail) [reply docs/user_guide/index.md:24]: That's true. Was just an initial idea. I think this explanation with each instrument is quite a lot and unnecessary for a new user. But we should definitely include a section to describe creating a new configuration.

And regarding jittering, I thought it's part of the standard workflow so would be good to be in basics, perhaps also a basic intro.

File: docs/user_guide/installation.md (hunk 8 above)
Comment (zain-sohail) [reply docs/user_guide/installation.md:74]: I moved that to the maintainer section 
https://opencompes.github.io/docs/sed/latest/misc/maintain.html

I can of course put it back but it seems most users don't want to use poetry. Perhaps linking that page in this file makes sense?

File: docs/workflows/index.md (hunk 9 above)
Comment (zain-sohail) [reply docs/workflows/index.md:14]: I think it makes sense to have an introduction to the different aspects of the library, irrespective of the different loaders/workflows at different instruments.
What is your suggestion?

File: docs/user_guide/index.md (hunk 6 above)
Comment (zain-sohail) [reply docs/user_guide/index.md:19]: I get the point. But I think we need to include more documentation about the library and the workflow tutorials are for specific cases.
For example, I wouldn't use the WSe2 entire dataset for jittering example but we can take a small subset of it and show how jittering works similar to tutorial 1.

File: docs/conf.py (hunk 4 above)
Comment (zain-sohail) [reply docs/conf.py:118]: See right side
image

Comment (rettigl) [reply docs/conf.py:118]: What I mean is that these menues are all distributed over the page, i.e. main menu on the top, section menue on the left, page menu on the right. I find this very confusing, and liked the unique tree of the old docs much better

File: docs/user_guide/installation.md (hunk 8 above)
Comment (rettigl) [reply docs/user_guide/installation.md:74]: That's probably fine.

File: docs/conf.py (hunk 4 above)
Comment (zain-sohail) [reply docs/conf.py:118]: Yes, but this seems to be how basically all modern projects in python are documented. See [numpy](https://numpy.org/doc/stable/index.html), [scipy](https://docs.scipy.org/doc/scipy/), [xarray](https://docs.xarray.dev/en/stable/), [dask](https://docs.dask.org/en/stable/). I think dask keeps the page menu on both left and right siden and some others might too. 
Since our library is based on these packages, I think it's more intuitive for user to follow such layout. We can of course change it but including complete heirarchy in the left menu clutters everything, I find.

Comment (rettigl) [reply docs/conf.py:118]: I can't follow. In the examples you provide, the left menu always stays the same, independent where you navigate. For your new layout, this is not the case. This is very confusing, and makes it difficult to get an overview of the docs. This is my point.

File: .github/workflows/documentation.yml (hunk 3 above)
Comment (rettigl) [reply .github/workflows/documentation.yml:119]: added lines to remove this code

File: docs/conf.py (hunk 4 above)
Comment (rettigl) [reply docs/conf.py:118]: We can probably keep the current layout, changing that appears to be not so straight-forward

File: docs/workflows/index.md (hunk 9 above)
Comment (rettigl) [reply docs/workflows/index.md]: I reordered things slightly, and added a section advanced topics

File: docs/user_guide/index.md (hunk 6 above)
Comment (rettigl) [reply docs/user_guide/index.md:19]: Restructured

**Review by rettigl — APPROVED** [review]
Fnished from my side

**Review by zain-sohail — COMMENTED** [review]
I have taken a look at the develop docs and looks good. Any doc changes, we can always do later. I'd say we merge and see if the switching is working properly. Otherwise we might need to delete the static files and rebuild them.
